test/external/guardian/liquidationWithdrawalKeyHijacking.ts

```diff
@@ -0,0 +1,518 @@
+      );
+
+      console.log(" 7. Balances for accounts after liquidations show no change to second account and first account has only liquidation output token");
+      const liquidAccountOneAfterBalances = await getMarketBalances(liquidAccount);
+      expect(liquidAccountOneAfterBalances.get(marketId.toString()) === undefined).to.be.false; // balance 0
+      expect(liquidAccountOneAfterBalances.get(core.marketIds.weth.toString()) === undefined).to.be.true; // balance 0
+      expect(liquidAccountOneAfterBalances.get(core.marketIds.nativeUsdc!.toString())?.isZero()).to.be.false; // liquidation output token was in USDC
+
+      const liquidAccountTwoAfterBalances = await getMarketBalances(liquidAccount2);
+      expect(liquidAccountTwoAfterBalances.get(marketId.toString())).to.be.equal(liquidAccountTwoBeforeBalances.get(marketId.toString()));
+      expect(liquidAccountTwoAfterBalances.get(core.marketIds.weth.toString())).to.be.equal(liquidAccountTwoBeforeBalances.get(core.marketIds.weth.toString()));
+      expect(liquidAccountTwoAfterBalances.get(core.marketIds.nativeUsdc!.toString())).to.be.equal(liquidAccountTwoBeforeBalances.get(core.marketIds.nativeUsdc!.toString()));
+
+      console.log(` 8. Since account ${liquidAccount.number} is now over-collateralized, the operation to retrieve his failed withdrawal can be initiated by a trusted handler`);
+      
+      await gmxV2Registry.connect(core.governance).ownerSetIsHandler(core.hhUser1.address, true);
+      const unwrapperAsTrustedHandler = unwrapper.connect(core.hhUser1);
+      const result = await unwrapperAsTrustedHandler.executeWithdrawalForRetry(firstBorrowerKey, {gasLimit: 30_000_000});
+      await result.wait();
+
+      console.log(" 9. But this operation will fail since adding the extra funds from the initial key would be interpreted by DolomiteMargin as an increase in borrowing from the GM market, which is a closed market");
+      await expectEvent(eventEmitter, result, 'AsyncWithdrawalFailed', {
+        key: firstBorrowerKey,
+        token: factory.address,
+        reason: `OperationImpl: Market is closing <${marketId}>`
+      });
+
+      console.log("10. The second position has also become over-collateralize during the price variation we did at (4) so continuing liquidation will fail regardless of withdrawal key");
+      console.log("11. If we try to liquidate using the second key (already used), even if the second account would be liquidatable, it reverts on the amount check since it is done first");
+      await expectThrow(
+        executeProxyLiquidation(
+          solidAccount, 
+          liquidAccount2, 
+          [secondBorrowerKey], 
+          [], 
+          _amountWeiForLiquidationTwo, 
+          _wethAmountTwo
+        ), 
+        "AsyncIsolationModeUnwrapperImpl: Invalid input amount"
+        );
+        await expectThrow(
+          executeProxyLiquidation(
+            solidAccount,
+            liquidAccount2,
+            [secondBorrowerKey],
```

Comment:
Same key is used as above.

Reply:
this is actually a duplicated call. Removing, the one showing the attempt using the first key is right at the end